### PR TITLE
Make dask arrays instead of computing slice for slice

### PR DESCRIPTION
**Pull request by Cryaaa**
Hey @haesleinhuepf!
this is the first implementation of the time slicer wrapper using dask instead of computing the time slices based on the current time index. I could re-use some a little of the previous code but the wrappers start to differ from eachother pretty soon. At the moment I'm also unsure if this wrapper can replace the original time slicer function as a substitute so I kept both your old version and the dask version. An idea that I had which could be useful for saving the dask images is a function which processes each time slice and saves it as a separate image (If images are saved one by one it's really easy to load them as dask arrays!)

### CODE REVIEW

File: src/napari_time_slicer/__init__.py [hunk 1]
```diff
@@ -51,6 +56,76 @@ def worker_function(*args, **kwargs):
...
+                    for delayed_reader in lazy_arrays
+                ]
+                # Stack into one large dask.array
+                stack = da.stack(
+                    dask_arrays, 
+                    axis=0)
+                return stack
+
+        # call the decorated function
+        result = function(*bound.args, **bound.kwargs)
```
Comment (haesleinhuepf) [inline src/napari_time_slicer/__init__.py:125]: Hi Ryan @Cryaaa ,

that's an amazing PR! Before I try it out, one quick question: I'm confused about this line. Are you sure it's necessary?

Big thanks!
Robert

Comment (Cryaaa) [reply src/napari_time_slicer/__init__.py:125]: Hey @haesleinhuepf 
I think this line is necessary if you still want to be able to apply the function you have wrapped to non-timelapse data. This functionality exists with your previous implementation and I did not want to break this functionality.

Comment (haesleinhuepf) [reply src/napari_time_slicer/__init__.py:125]: Ok, I'll try it out then. Big thanks! That's a super cool feature!

**Comment by haesleinhuepf** [comment]
Big thanks Ryan @Cryaaa !

**Pull request merged**